src/kernel.ts

```diff
@@ -434,7 +477,14 @@ class Kernel {
   }
 
   /**
-   * Create a Kernel Message given input attributes.
+   * Get the current kernel url.
+   */
+  private get _kernelUrl(): string {
+    return utils.urlJoinEncode(this._baseUrl, KERNEL_SERVICE_URL, this._id);
```

Comment:
Should this be computed once in the constructor as a normal private member? Do `_baseUrl` and `_id` ever change?